app/controllers/application_controller.rb

```diff
@@ -3,4 +3,6 @@
 class ApplicationController < ActionController::Base
   include Localization
   include Pagy::Backend
+
+  before_action :authenticate_user!
```

Comment:
I think we need to check the unit test and feature test due to this changes as we would need to login in all screens - upload, view details,...